python/cog/server/redis_queue.py

```diff
@@ -186,15 +209,19 @@ def handle_message(self, response_queue, message, cleanup_functions):
             self.push_error(response_queue, e)
             return
 
-        with self.capture_log(self.STAGE_RUN, prediction_id):
+        with self.capture_log(self.STAGE_RUN, prediction_id), timeout(
+            seconds=self.predict_timeout
+        ):
             return_value = self.predictor.predict(**inputs)
         if isinstance(return_value, types.GeneratorType):
             last_result = None
 
             while True:
                 # we consume iterator manually to capture log
                 try:
-                    with self.capture_log(self.STAGE_RUN, prediction_id):
+                    with self.capture_log(self.STAGE_RUN, prediction_id), timeout(
```

Comment:
Doesn't this mean that timeouts are applied per-yield? So the whole prediction can take `n_yields * timeout` seconds?

Reply:
Correct, thanks for the fix!